### PR TITLE
Mldb 1181 - add an option to not push empty rows

### PR DESCRIPTION
**Pull request by steadtler**
In the transform procedure.

### CODE REVIEW

File: plugins/sql_functions.cc [hunk 1]
```diff
@@ -484,13 +490,16 @@ run(const ProcedureRunConfig & run,
                     cols.push_back(c);
                 }
 
-                auto & rows = accum.get();
-                rows.reserve(10000);
-                rows.emplace_back(RowName(calc.at(0).toString()), std::move(cols));
+                if (!skipEmptyRows || cols.size() > 0)
```
Comment (guyd) [inline plugins/sql_functions.cc]: Are you sure this is the condition you want?

**Comment by guyd** [comment]
+1

**Comment by nicolaskruchten** [comment]
Hmmm I'm not sure I like the process used in this PR... There was no DOD for this option. Why did we need a new option for this as opposed to just doing it within the SQL query of the transform?

**Comment by steadtler** [comment]
We found no elegant and performant way of doing this - it either takes two queries and some custom code from the client, or one much more expensive query. You weren't there so I checked with Frank beforehand.

**Comment by nicolaskruchten** [comment]
The "SQL way" to do this, to my mind, would be to repeat the `select` expression in the `where` and for MLDB to cache the selection to reduce the expense. That said, this is acceptable for now.

**Pull request merged**